conda/cli/main_search.py

```diff
@@ -1,16 +1,18 @@
-# (c) 2012-2013 Continuum Analytics, Inc. / http://continuum.io
-# All Rights Reserved
-#
-# conda is distributed under the terms of the BSD 3-clause license.
-# Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
-
+# -*- coding: utf-8 -*-
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-from os.path import dirname
+import sys
+from argparse import SUPPRESS
+from collections import defaultdict
 
+from conda.core.index import get_channel_priority_map
 from .conda_argparse import (add_parser_channels, add_parser_insecure, add_parser_json,
                              add_parser_known, add_parser_offline, add_parser_prefix,
                              add_parser_use_index_cache, add_parser_use_local)
+from ..cli.common import stdout_json
+from ..common.io import spinner
+from ..compat import itervalues
+from ..exceptions import PackagesNotFoundError
 
 descr = """Search for packages and display their information. The input is a
```

Comment:
We need to update these `descr` and `example` strings.  Go ahead and take a first crack at it.  Remove anything that's no longer relevant.  Also, come up with some good examples based on the new `MatchSpec`.  Look at the docstring on `MatchSpec` for examples there.